src/net.c

```diff
@@ -275,8 +270,8 @@ void mg_mgr_init(struct mg_mgr *mgr) {
   // Ignore SIGPIPE signal, so if client cancels the request, it
   // won't kill the whole process.
   signal(SIGPIPE, SIG_IGN);
-#elif MG_ENABLE_TCPIP && MG_ENABLE_TCPIP_DRIVER_INIT
-  mg_tcpip_auto_init(mgr);
+#elif MG_ENABLE_TCPIP_DRIVER_INIT && defined(MG_TCPIP_DRIVER_INIT)
+  MG_TCPIP_DRIVER_INIT(mgr);
 #endif
```

Comment:
I don't think this should be here, it is strictly for MIP and so IMHO I think it belongs in net_builtin,c

Reply:
Why? What's the point of adding a function what calls that macro?
What's the benefit of having extra code?

Reply:
Clarity ? Keeping stacks separate ? I see net.c as a layer for a generic interface with the underlying specific protocol stack. That clearly does not have drivers. Drivers belong to protocol stacks and should be initialized by protocol stacks.
We add another protocol stack, I don't know, some Matter-like, that will have its drivers and those will need initialization, that will happen in net_someothernetstack.c, for example.

Reply:
IMO adding more indirection here, would be less clear.
With those three lines, you see EXACTLY what is going on, and why.

What's the reason to add same three lines but calling some mg_tcpip_whatever function?

Reply:
They wouldn't be the same lines...
```
#elif MG_ENABLE_TCPIP           // if we are using our own stack, init it
  mg_tcpip_init();
#elif MG_ENABLE_WHATEVERSTACK   // following up on the example above
  mg_whateverstack_init();
#endif
```
That is all net.c needs to know. It does not know nor cares about drivers and their initialization
Now, net_builtin.c does know about MG_ENABLE_TCPIP_DRIVER_INIT, it is TCPIP-land, and it knows about drivers and stuff, so those famous three lines would be in net_builtin.c, inside  mg_tcpip_init()

Anyway, this is philosophical... for the sake of efficiency we should write the Ethernet driver here, and for the sake of separation we should have a gazillion modules and take forever to call each one... You surely know better where to draw the line.